lib/chef/resource/cron/_cron_shared.rb

```diff
@@ -0,0 +1,102 @@
+unified_mode true
+
+WEEKDAYS = {
+  sunday: "0", monday: "1", tuesday: "2", wednesday: "3", thursday: "4", friday: "5", saturday: "6",
+  sun: "0", mon: "1", tue: "2", wed: "3", thu: "4", fri: "5", sat: "6"
+}.freeze
+
+property :minute, [Integer, String],
+  description: "The minute at which the cron entry should run (`0 - 59`).",
+  default: "*", callbacks: {
+    "should be a valid minute spec" => ->(spec) { Chef::ResourceHelpers::CronValidations.validate_numeric(spec, 0, 59) },
+  }
+
+property :hour, [Integer, String],
+  description: "The hour at which the cron entry is to run (`0 - 23`).",
+  default: "*", callbacks: {
+    "should be a valid hour spec" => ->(spec) { Chef::ResourceHelpers::CronValidations.validate_numeric(spec, 0, 23) },
+  }
+
+property :day, [Integer, String],
+  description: "The day of month at which the cron entry should run (`1 - 31`).",
+  default: "*", callbacks: {
+    "should be a valid day spec" => ->(spec) { Chef::ResourceHelpers::CronValidations.validate_numeric(spec, 1, 31) },
```

Comment:
We should be able to move all the validation files out of the helper file and into this shared file. The format changes a bit, but if you look back in the history you can find where I pulled that into the shared file. It would be nice to kill off that helper and just include all those methods here. That will also require moving around some rspecs most likely.

Reply:
@tas50 I tried to find  in the history but I was not able to see anything related to helper.Do I need to move all the methods in `cron_validation` helper to the shared partial?

Reply:
You can just try to move the helpers into this resource partial and use them directly. There's no need to have a partial and the helpers.